pkgs/games/steam/fhsenv.nix

```
@@ -46,14 +51,12 @@ let
     sqlite
   ] ++ extraPkgs pkgs;
 
-  ldPath = lib.optionals stdenv.is64bit [ "/lib64" ]
-  ++ [ "/lib32" ]
-  ++ map (x: "/steamrt/${steam-runtime-wrapped.arch}/" + x) steam-runtime-wrapped.libs
-  ++ lib.optionals (steam-runtime-wrapped-i686 != null) (map (x: "/steamrt/${steam-runtime-wrapped-i686.arch}/" + x) steam-runtime-wrapped-i686.libs);
-
+  # Prepend /lib64 and /lib32 to LD_LIBRARY_PATH so Nixpkgs tools running from Steam
+  # don't end up choosing Steam Runtime libraries, because LD_LIBRARY_PATH beats
+  # DT_RUNPATH. This is a horrible hack.
   # Zachtronics and a few other studios expect STEAM_LD_LIBRARY_PATH to be present
   exportLDPath = ''
-    export LD_LIBRARY_PATH=${lib.concatStringsSep ":" ldPath}''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
+    export LD_LIBRARY_PATH=/lib64:/lib32:$LD_LIBRARY_PATH
```

I think you probably still want the `''${LD_LIBRARY_PATH:+:}` part. Otherwise, it's possible for this to become `LD_LIBRARY_PATH=:/lib64:/lib32:` (note the empty segment at the end), which could cause issues for some things.
```suggestion
    export LD_LIBRARY_PATH=/lib64:/lib32''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
```